bundles/org.openhab.binding.haywardomnilogic/src/main/java/org/openhab/binding/haywardomnilogic/internal/discovery/HaywardDiscoveryService.java

```diff
@@ -0,0 +1,454 @@
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.haywardomnilogic.internal.discovery;
+
+import static org.openhab.binding.haywardomnilogic.internal.HaywardBindingConstants.THING_TYPES_UIDS;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.haywardomnilogic.internal.HaywardBindingConstants;
+import org.openhab.binding.haywardomnilogic.internal.HaywardException;
+import org.openhab.binding.haywardomnilogic.internal.HaywardTypeToRequest;
+import org.openhab.binding.haywardomnilogic.internal.handler.HaywardBridgeHandler;
+import org.openhab.core.config.discovery.AbstractDiscoveryService;
+import org.openhab.core.config.discovery.DiscoveryResult;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.config.discovery.DiscoveryService;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Sets up the discovery results and details
+ *
+ * @author Matt Myers - Initial contribution
+ */
+
+@NonNullByDefault
+public class HaywardDiscoveryService extends AbstractDiscoveryService implements DiscoveryService, ThingHandlerService {
+    private final Logger logger = LoggerFactory.getLogger(HaywardDiscoveryService.class);
+    private @Nullable HaywardBridgeHandler discoveryBridgehandler;
+
+    public HaywardDiscoveryService() {
+        super(THING_TYPES_UIDS, 0, false);
+    }
+
+    /**
+     * Constructs a zone discovery service.
+     * Registers this zone discovery service programmatically.
+     * Call {@link ZoneDiscoveryService#destroy()} to unregister the service after use.
+     */
+    public HaywardDiscoveryService(HaywardBridgeHandler bridge) throws IllegalArgumentException {
+        super(THING_TYPES_UIDS, 0, false);
+    }
+
+    @Override
+    public void activate() {
+        super.activate(null);
+    }
+
+    @Override
+    public void deactivate() {
+        super.deactivate();
+    }
+
+    @Override
+    protected void startScan() {
+        HaywardBridgeHandler bridgehandler = discoveryBridgehandler;
+        try {
+            if (bridgehandler != null) {
+                String xmlResults = bridgehandler.getMspConfig();
+                mspConfigDiscovery(xmlResults);
+            }
+        } catch (HaywardException e) {
+            logger.warn("Exception during discovery scan: {}", e.getMessage());
+        } catch (InterruptedException e) {
+            return;
+        }
+    }
+
+    public synchronized boolean mspConfigDiscovery(String xmlResponse) throws HaywardException {
```

Comment:
This exception is never thrown.
```suggestion
    public synchronized boolean mspConfigDiscovery(String xmlResponse) {
```